pysensors/optimizers/_gqr.py

```diff
@@ -71,44 +69,12 @@ def fit(self,basis_matrix=None,**optimizer_kws):
 
         Returns
         -------
-        self: a fitted :class:`pysensors.optimizers.QR` instance
+        self: a fitted :class:`pysensors.optimizers.GQR` instance
         """
-        if 'idx_constrained' in optimizer_kws.keys():
-            self.constrainedIndices = optimizer_kws['idx_constrained']
-        else:
-            self.constrainedIndices = []
-        if 'n_sensors' in optimizer_kws.keys():
-            self.n_sensors = optimizer_kws['n_sensors']
-        else:
-            self.n_sensors = np.shape(basis_matrix)[0]
-        if 'n_const_sensors' in optimizer_kws.keys():
-            self.nConstrainedSensors = optimizer_kws['n_const_sensors']
-        else:
-            self.nConstrainedSensors = None
-        if 'all_sensors' in optimizer_kws.keys():
-            self.all_sensors = optimizer_kws['all_sensors']
-        else:
-            self.all_sensors = None
-        if 'constraint_option' in optimizer_kws.keys():
-            self.constraint_option = optimizer_kws['constraint_option']
-            self._norm_calc_Instance = normCalcReturnInstance(self,self.constraint_option)
-        else:
-            self.constraint_option = None
-        if 'nx' in optimizer_kws.keys():
-            self._nx = optimizer_kws['nx']
-        else:
-            self._nx = None
-        if 'ny' in optimizer_kws.keys():
-            self._ny = optimizer_kws['ny']
-        else:
-            self._ny = None
-        if 'r' in optimizer_kws.keys():
-            self._r = optimizer_kws['r']
-        else:
-            self._r = None
-
+        [setattr(self,name,optimizer_kws.get(name,getattr(self,name))) for name in optimizer_kws.keys()]
+        self._norm_calc_Instance = normCalcReturnInstance(self, self.constraint_option)
         n_features, n_samples = basis_matrix.shape  # We transpose basis_matrix below
-        max_const_sensors = len(self.constrainedIndices) # Maximum number of sensors allowed in the constrained region
+        max_const_sensors = len(self.idx_constrained) # Maximum number of sensors allowed in the constrained region
 
         ## Assertions and checks:
```

Comment:
We could cut this out for now as I think we had added this but it was giving us errors with SVD basis.